### PR TITLE
Feature us2781 mailchimp opt ins

### PR DESCRIPTION
**Pull request by danrye**

### CODE REVIEW

File: CI/SQL/US2781-01-Mail-Chimp-PageViews.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+USE MinistryPlatform
+GO
+
+SET IDENTITY_INSERT dbo.dp_Page_Views ON
+
```
Comment (silbermm) [inline CI/SQL/US2781-01-Mail-Chimp-PageViews.sql]: Check if the table exists first!

File: Gateway/crds-angular/Services/BulkEmailSyncService.cs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+                return null;
+            }
+        }
+
+        private void LogUpdateStatuses(Dictionary<string, string> publicationOperationIds)
+        {
+            // poll mailchimp to see if batch was successful -- also need to confirm during dev
+            // testing if the response is synchronous or asynchronous -- 99% sure it's synchronous
+            var client = new RestClient("https://us12.api.mailchimp.com/3.0/");
```
Comment (silbermm) [inline Gateway/crds-angular/Services/BulkEmailSyncService.cs]: should the url be in the configuration file

File: Gateway/crds-angular/Services/BulkEmailSyncService.cs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+                //TODO: Determine how to populate the ThirdPartyPublicationID? For now you may just write SQL to update table directly
+                mailChimpSubscriber.path = string.Format("lists/{0}/members/{1}", publication.ThirdPartyPublicationId, hashedEmail);
+
+                // TODO: Do we need to store this somewhere to verify batch processed successfully
+                mailChimpSubscriber.operation_id = Guid.NewGuid().ToString();
+
+                string mergeFields = JsonConvert.SerializeObject(subscriber.MergeFields);
+                // TODO: Use JSON.NET to serialize this rather than a string
+                mailChimpSubscriber.body = String.Format("{{ \"status\":\"{0}\", \"email_address\":{1}, \"merge_fields\":{2} }}",
```
Comment (silbermm) [inline Gateway/crds-angular/Services/BulkEmailSyncService.cs]: use a JSON Serializer

File: CI/SQL/US2781-01-Mail-Chimp-PageViews.sql (hunk 1 above)
Comment (jpc11623) [inline CI/SQL/US2781-01-Mail-Chimp-PageViews.sql]: Fixed script

**Pull request merged**